From c1c8509bb90f2b3ed046d99cf0fedbddc24290ec Mon Sep 17 00:00:00 2001
From: Myy Miouyouyou <myy@miouyouyou.fr>
Date: Thu, 3 May 2018 21:47:40 +0200
Subject: [PATCH] drivers: drm: rockchip: Enable IRQ on unbind

Not doing this generate atrocious delays when plugging a screen,
making the whole 3D stack unusable.

This is an adaptation of Jeffy Chen patch, originally provided
here :
http://lists.infradead.org/pipermail/linux-rockchip/2018-April/020427.html

Thanks to @JeffyCN for this patch.

This resolves issue #4

Signed-off-by: Myy Miouyouyou <myy@miouyouyou.fr>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
index 53d4afe1..f903171a 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
@@ -1601,6 +1601,7 @@ static void vop_unbind(struct device *dev, struct device *master, void *data)
 {
 	struct vop *vop = dev_get_drvdata(dev);
 
+	enable_irq(vop->irq);
 	pm_runtime_disable(dev);
 	vop_destroy_crtc(vop);
 
-- 
2.17.0

